### PR TITLE
Minor improvements

### PR DESCRIPTION
**Pull request by sultaniman**
A small refactoring and minor conditional flow changes.
Also added checks if the input `isArray` otherwise to take `arguments`.

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -1,18 +1,16 @@
 'use strict';
-module.exports = function (arr) {
-	if (!Array.isArray(arr)) {
-		throw new TypeError('Expected an array to multiply its items');
-	}
+module.exports = function(arr) {
```
Comment (kevva) [inline index.js:2]: `function (arr)`

**Comment by arthurvr** [comment]
> A small refactoring and minor conditional flow changes.

I like it early-exiting instead of an extra if-else statement. It's simpler and more explicit.

> Also added checks if the input isArray otherwise to take arguments.

Personally I don't like it taking `arguments` instead of an array. It's rarely useful and I don't like the API complication. WDYT?

> var product = 1;

Not sure about this. Isn't starting with the first item more logical?

**Comment by kevva** [comment]
Why would it accept `arguments` if the module is named `multiply-array` :)?

**Comment by sultaniman** [comment]
@kevva users usually do pass multiple arguments I guess :smile:

**Comment by sultaniman** [comment]
@arthurvr 

> Not sure about this. Isn't starting with the first item more logical?

Yes I agree

> I like it early-exiting instead of an extra if-else statement. It's simpler and more explicit.

Then can we just return the first item in the list instead throwing exception?

**Comment by sultaniman** [comment]
@arthurvr @kevva I'll make related changes as you advised

**Comment by arthurvr** [comment]
Please also remove it taking `arguments`. The only change I really like is `*=`.

> Then can we just return the first item in the list instead throwing exception?

No, I like it throwing. I don't see why not.

**Comment by sultaniman** [comment]
@arthurvr sure I will revert then

**Comment by arthurvr** [comment]
>  The only change I really like is *=.

I still see way more than that. Did it myself now :wink:

**Comment by sultaniman** [comment]
@arthurvr cool thank you :wink:

**Pull request closed without merging**